### PR TITLE
NAY4-2_revert_reward_payment_if_staking_not_initialised

### PR DESCRIPTION
**Pull request by Huanzhang89**

### CODE REVIEW

File: src/libs/LibTokenizedVaultStaking.sol [hunk 1]
```diff
@@ -82,6 +82,10 @@ library LibTokenizedVaultStaking {
     function _payReward(bytes32 _stakingRewardId, bytes32 _entityId, bytes32 _rewardTokenId, uint256 _rewardAmount) internal {
         AppStorage storage s = LibAppStorage.diamondStorage();
 
+        if (s.stakingConfigs[_entityId].initDate == 0) {
```
Comment (amarinkovic) [inline src/libs/LibTokenizedVaultStaking.sol:85]: This is what we said we will do, if I remember correctly 🤔 
```suggestion
        if (s.stakingConfigs[_entityId].initDate >= block.timestamp) {
```

Comment (amarinkovic) [inline src/libs/LibTokenizedVaultStaking.sol:85]: Isn't this what we want? 🤔 
```suggestion
        if (s.stakingConfigs[_entityId].initDate <= block.timestamp) {
```

Comment (amarinkovic) [inline src/libs/LibTokenizedVaultStaking.sol:100]: Isn't this what we want? 🤔 
```suggestion
        if (s.stakingConfigs[_entityId].initDate == 0 || s.stakingConfigs[_entityId].initDate > block.timestamp) {
```

Comment (Huanzhang89) [reply src/libs/LibTokenizedVaultStaking.sol:100]: > Isn't this what we want? 🤔

umm there is another check a few lines down for this which returns a more specific error. I think its better to separate these checks as we want the error messages to be clear on why the tx was reverted

Comment (amarinkovic) [reply src/libs/LibTokenizedVaultStaking.sol:100]: Good point, ignore my comment.

**Review by amarinkovic — APPROVED** [review]

**Pull request merged**